src/main/java/com/github/sylvainlaurent/maven/yamljsonvalidator/ValidateMojo.java

```diff
@@ -111,23 +113,37 @@ public void execute() throws MojoExecutionException {
 
     InputStream openJsonSchema(String jsonSchemaFile) throws MojoExecutionException {
         if (jsonSchemaFile != null && jsonSchemaFile.length() > 0) {
-            File file = new File(jsonSchemaFile);
-            if (file.isFile()) {
-                try {
-                    return new FileInputStream(file);
-                } catch (FileNotFoundException e) {
-                    throw new MojoExecutionException("Could not load schema file ["+ jsonSchemaFile + "]", e);
-                }
-            } else {
-                try {
-                    InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream(jsonSchemaFile);
-                    if (inputStream != null) {
-                        return inputStream;
+            try {
+                URI uri = new URI(jsonSchemaFile);
+                if(uri.getScheme() == null) {
+                    File file = new File(jsonSchemaFile);
+                    if (file.isFile()) {
+                        try {
+                            return new FileInputStream(file);
+                        } catch (FileNotFoundException e) {
+                            throw new MojoExecutionException("Could not load schema file ["+ jsonSchemaFile + "]", e);
+                        }
+                    } else {
+                        try {
+                            InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream(jsonSchemaFile);
+                            if (inputStream != null) {
+                                return inputStream;
+                            }
+                            throw new MojoExecutionException("Could not load schema neither from filesystem nor classpath ["+ jsonSchemaFile + "]");
+                        } catch (Exception e){
+                            throw new MojoExecutionException("Could not load schema file from classpath ["+ jsonSchemaFile + "]", e);
+                        }
+                    }
+                } else {
+                    try (final CloseableHttpClient httpclient = HttpClients.custom().useSystemProperties().build()) {
+                        HttpGet get = new HttpGet();
+                        get.setURI(uri);
+                        return httpclient.execute(get).getEntity().getContent();
```

Comment:
`execute(get)` here returns `CloseableHttpResponse`. It should be wrapped in try-with-resource as well.

Reply:
Thank you @wilx , I do it.